lib/plugins/aws/lib/naming.js

```diff
@@ -545,8 +545,13 @@ module.exports = {
   getLambdaAtEdgeInvokePermissionLogicalId(functionName) {
     return `${this.getLambdaLogicalId(functionName)}InvokePermission`;
   },
-  getCloudFrontOriginId(functionName, path) {
-    let originId = this.getNormalizedFunctionName(functionName);
+  getCloudFrontOriginId(isS3Origin, domain, path) {
```

Comment:
It'll probably be nicer to change signature simply to `originObj` and resolve needed values from it.